library/system/src/lib/yast2/fs_snapshot.rb

```diff
@@ -0,0 +1,152 @@
+      super "Snapper could not be configured."
+    end
+  end
+
+  # Class for managing filesystem snapshots. It's important to note that this
+  # class is intended to be used during installation/update so it uses the
+  # Snapper's CLI because the DBus interface is not available at that time.
+  class FsSnapshot
+    include Yast::Logger
+
+    FIND_CONFIG_CMD = "/usr/bin/snapper --no-dbus list-configs | grep \"^root \" >/dev/null"
+    CREATE_CONFIG_CMD = "/usr/bin/snapper --no-dbus create-config -f btrfs /"
+    CREATE_SNAPSHOT_CMD = "/usr/lib/snapper/installation-helper --step 5 --description \"%s\""
+    LIST_SNAPSHOTS_CMD = "LANG=en_US.UTF-8 /usr/bin/snapper --no-dbus list"
+    VALID_LINE_REGEX = /\A\w+\s+\| \d+/
+
+    attr_reader :number, :snapshot_type, :previous_number, :timestamp, :user,
+      :cleanup_algo, :description
+
+    # Determines whether snapper is configured or not
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configured?
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), FIND_CONFIG_CMD)
+      log.info("Checking if Snapper is configured: \"#{FIND_CONFIG_CMD}\" returned: #{out}")
+      out["exit"] == 0
+    end
+
+    # Configures snapper
+    #
+    # @return [true,false] true if it's configured; false otherwise.
+    def self.configure
+      unless configured?
+        out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), CREATE_CONFIG_CMD)
+        unless out["exit"] == 0
+          log.error("Snapper configuration failed: #{CREATE_CONFIG_CMD} returned: #{out}")
+          raise SnapperConfigurationFailed
+        end
+      end
+      true
+    end
+
+    # Creates a new snapshot
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @param description [String] Snapshot's description.
+    # @return [FsSnapshot,nil] The created snapshot if the operation was
+    #                          successful. Otherwise, it returns nil.
+    def self.create(description)
+      raise SnapperNotConfigured unless configured?
+
+      cmd = CREATE_SNAPSHOT_CMD % description
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), cmd)
+      if out["exit"] == 0
+        find(out["stdout"].to_i) # The CREATE_SNAPSHOT_CMD returns the number of the new snapshot.
+      else
+        log.error "Snapshot could not be created: #{CREATE_SNAPSHOT_CMD} returned: #{out}"
+      end
+    end
+
+    # Returns all snapshots
+    #
+    # It raises and exception if Snapper is not configured.
+    #
+    # @return [Array<FsSnapshot>] All snapshots that exist in the system.
+    def self.all
+      raise SnapperNotConfigured unless configured?
+
+      out = Yast::SCR.Execute(Yast::Path.new(".target.bash_output"), LIST_SNAPSHOTS_CMD)
+      lines = out["stdout"].lines.grep(VALID_LINE_REGEX) # relevant lines from output.
+      log.info("Retrieving snapshots list: #{LIST_SNAPSHOTS_CMD}. Found #{lines.size} snapshot(s).")
```

Comment:
ok, consider that there something changed in output and you get 0 snapshots, this line won't help you much. Usually it is good idea to capture input or output on "borders" which is in this case CLI. So I think it is better to log output itself.